components/org.wso2.carbon.appmgt.mobile/src/main/java/org/wso2/mobile/utils/utilities/ZipFileReading.java

```diff
@@ -109,8 +109,14 @@ public String readiOSManifestFile(String filePath, String name) {
                         break;
                     }
                 }
-                NSDictionary rootDict = (NSDictionary) BinaryPropertyListParser
-                        .parse(buffer.toByteArray());
+                NSDictionary rootDict;
+                try{
+                    rootDict = (NSDictionary) BinaryPropertyListParser
+                            .parse(buffer.toByteArray());
+                }catch(IllegalArgumentException e){
+                    log.debug("Uploaded file didn't have a Binary Plist");
+                    rootDict = (NSDictionary) PropertyListParser.parse(buffer.toByteArray());
```

Comment:
Need to add a try-catch to catch any exceptions thrown while parsing plain text configs and log them as well.